x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts

```diff
@@ -108,7 +108,10 @@ describe('onPostAuthInterceptor', () => {
     availableSpaces: any[],
     testOptions = { simulateGetSpacesFailure: false, simulateGetSingleSpaceFailure: false }
   ) {
-    const { http } = await root.setup();
+    const { http, elasticsearch } = await root.setup();
+
+    // Mock esNodesCompatibility$ to prevent `root.start()` from blocking on ES version check
+    elasticsearch.esNodesCompatibility$ = elasticsearchServiceMock.createInternalSetup().esNodesCompatibility$;
```

Comment:
Any reason not to do this in KbnTestUtils? Seems like this could break other tests in confusing ways in the future

Reply:
Yes, I find this rather ugly. KbnTestUtils is sometimes used with an ES server in which case we don't need to fake `esNodesCompatibility$`, but plugins should never have to run tests against an incompatible ES node so it's probably safe to always skip this during testing.

Reply:
To prevent potentially long build/test cycle when changing integration tests, I'll rather do this in a separate PR.